### PR TITLE
replace NewClient with NewRegistryClient

### LINKED ISSUE #660: pkg/connection: let's deprecate NewClient and NewClientWithSettings and add NewRegistryClient and NewRegistryClientWithSettings
https://github.com/apigee/registry/blob/7bc06ba752571bcb29b327c4351e0a810cc9e283/pkg/connection/client.go#L74

### PR DESCRIPTION
**Pull request by theganyo**
Fixes #660

### CODE REVIEW

**Review by timburks — APPROVED** [review]

File: gapic/doc.go [hunk 1]
```diff
@@ -59,7 +59,7 @@
 //
 // Use of Context
 //
-// The ctx passed to NewClient is used for authentication requests and
+// The ctx passed to NewRegistryClient is used for authentication requests and
```
Comment (timburks) [inline gapic/doc.go:62]: This is generated code, and this change gets clobbered when we rerun ./tools/GENERATE-GAPIC.sh. I think it's an upstream problem in the generator.

Comment (theganyo) [reply gapic/doc.go:62]: Should we merge and file a new issue on this?

Comment (theganyo) [reply gapic/doc.go]: rolled back the doc change

**Comment by timburks** [comment]
I filed [this issue](https://github.com/googleapis/gapic-generator-go/issues/1077) about the `doc.go` problem.

**Pull request merged**